regression/tests/lms/test_discussions.py

```diff
@@ -1,47 +1,104 @@
+        self.comment = 'comment{}'.format(uuid4().hex)
 
         try:
-            discussion_page.visit()
+            self.discussion_home_page_ext.visit()
         except PageLoadError:
-            discussion_page.url = '{}://{}/courses/{}/{}/{}/discussion/forum/'.format(
-                LMS_PROTOCOL,
-                LMS_BASE_URL,
-                COURSE_ORG,
-                COURSE_NUMBER,
-                COURSE_RUN
-            )
-            discussion_page.visit()
-
-        discussion_page.wait_for_page()
-        while discussion_page.is_browser_on_page():
-            break
+            raise unittest.SkipTest('The Discussions page isnt available')
+        self.discussion_home_page_ext.is_browser_on_page()
+
+    def test_discussion_page(self):
+        """
+        Tests discussions page:
+        1. Post can be added/deleted
+        2. Response to the post can be added/deleted
+        3. Comment to the response can be added/deleted
+        4. Test following
+        5. Test Search
+        """
+        #add new post
+        self.discussion_home_page_ext.click_new_post_button()
+        self.discussion_home_page_ext.fill_post_form(self.post_title, self.description)
+
+        #search_for_created post
+        self.discussion_home_page_ext.perform_search(self.post_title)
+        time.sleep(5)
+        link = self.discussion_home_page_ext.get_the_thread_link()
+
+        #visit thread page
+        self.discussion_thread_page_ext = DiscussionThreadPageExtended(
+            self.browser, get_course_key(get_course_info()), link=link)
+        self.discussion_thread_page_ext.visit()
+
+        #ASSERT that user can open created discussion
+        post_title = self.discussion_thread_page_ext.get_post_title()
+        self.assertEqual(''.join(post_title.text), self.post_title)
+
+        #check that post is followed
+        last_follow_post = self.discussion_thread_page_ext.last_follow_post()
+        self.assertEqual(last_follow_post, self.post_title)
+
+        #add a response
+        self.discussion_thread_page_ext.add_response(self.response)
+
+        #assert added reponse
+        response_text = self.discussion_thread_page_ext.get_response_text()
+        self.assertEqual(self.response, ''.join(response_text))
+
+        #add comment
+        self.discussion_thread_page_ext.add_comment(self.comment)
+        time.sleep(3)
+
+        #assert_comment_is_visible
+        self.assertTrue(self.discussion_thread_page_ext.is_comment_visible())
+
+        #delete comment
+        self.discussion_thread_page_ext.delete_comment()
+
+        #is_comment deleted
+        self.discussion_thread_page_ext.is_comment_deleted()
+
+        #delete response
+        self.discussion_thread_page_ext.delete_response()
+        self.discussion_thread_page_ext.is_response_deleted()
+
+        #delete post
+        try:
+            self.discussion_thread_page_ext.delete_post()
+            time.sleep(3)
+        except WrongPageError:
```

Comment:
@DenisKolesnik I propose to do `self.discussion_home_page_ext.is_post_deleted()` only in finally case.